src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -91,13 +102,164 @@ static void CreateHashDistributedTable(Oid relationId, char *distributionColumnN
+ *
+ */
+static uint32
+SupportFunctionForCitusSize(text *relationText, CitusSizeFunctionTypes sizeFunctionType)
+{
+	Oid distributedRelationId = ResolveRelationId(relationText);
+	DistTableCacheEntry *distributedTableCacheEntry = NULL;
+
+	/* Build shard qualified name. */
+	Oid schemaId = get_rel_namespace(distributedRelationId);
+	char *schemaName = get_namespace_name(schemaId);
+
+	int shardIndex = 0;
+	uint32 totalRelationSize = 0;
+
+	MultiConnection *connection = NULL;
+
+	distributedTableCacheEntry = DistributedTableCacheEntry(distributedRelationId);
+
+	for (shardIndex = 0; shardIndex <
+		 distributedTableCacheEntry->shardIntervalArrayLength; shardIndex++)
+	{
+		ShardPlacement *placement =
+			distributedTableCacheEntry->arrayOfPlacementArrays[shardIndex];
+		int64 shardId = placement->shardId;
+
+		connection = GetNodeConnection(0, placement->nodeName, placement->nodePort); /* TODO : I'm not sure about the flag ! */
+
+		char *shardName = get_rel_name(distributedRelationId);
+		AppendShardIdToName(&shardName, shardId);
+
+		char *shardQualifiedName = quote_qualified_identifier(schemaName, shardName);
+		char *quotedShardName = quote_literal_cstr(shardQualifiedName);
+
+		StringInfo tableSizeQuery = makeStringInfo();
+		PGresult *queryResult = NULL;
+
+		bool cstoreTable = CStoreTable(distributedRelationId);
+		if (cstoreTable)
+		{
+			appendStringInfo(tableSizeQuery, SHARD_CSTORE_TABLE_SIZE_QUERY,
+							 quotedShardName);
+		}
+		else
+		{
+			if (sizeFunctionType == CITUS_TOTAL_RELATION_SIZE)
+			{
+				appendStringInfo(tableSizeQuery, SHARD_TOTAL_RELATION_SIZE_QUERY,
+								 quotedShardName);
+			}
+			else if (sizeFunctionType == CITUS_TABLE_SIZE)
+			{
+				appendStringInfo(tableSizeQuery, SHARD_TABLE_SIZE_QUERY, quotedShardName);
+			}
+			else if (sizeFunctionType == CITUS_RELATION_SIZE)
+			{
+				appendStringInfo(tableSizeQuery, SHARD_RELATION_SIZE_QUERY,
+								 quotedShardName);
+			}
+		}
+
+		int executeCommand = ExecuteOptionalRemoteCommand(connection,
+														  tableSizeQuery->data,
+														  &queryResult);
+		/* If there is a connection error, close the connection and quit. */
+		if (executeCommand != 0)
+		{
+			PQclear(queryResult);
+
+			ForgetResults(connection);
+
+			ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+							errmsg("cannot get the size"),
+							errhint("Provide connection to each node then try again.")));
+		}
+
+		char *tableSizeString = PQgetvalue(queryResult, 0, 0);
+		totalRelationSize += atoi(tableSizeString);
+
+		PQclear(queryResult);
+
+		ForgetResults(connection);
+	}
+
+	return totalRelationSize;
+}
+
+
+/*
+ * citus_total_relation_size accepts a table name and
+ * returns a distributed table and its indexes' total relation size.
+ *
+ */
+Datum
+citus_total_relation_size(PG_FUNCTION_ARGS)
+{
+	text *relationText = PG_GETARG_TEXT_P(0);
+	uint32 totalRelationSize = 0;
+
+	ErrorIfNotSuitableToGetSize(relationText);
```

Comment:
We call in each citus_*_size UDFs. Maybe we can move it to the inside of the SupportFunctionForCitusSize?